dakara_server/library/management/commands/createworks.py

```diff
@@ -0,0 +1,225 @@
+        work_subtitles = dict_work['subtitles'] if 'subtitles' in dict_work else [""] # noqa E501
+
+        # get or create works
+        for work_subtitle in work_subtitles:
+            work_entry, work_created = Work.objects.get_or_create(
+                title__iexact=work_title,
+                work_type__id=work_type_entry.id,
+                subtitle__iexact=work_subtitle,
+                defaults={
+                    'title': work_title,
+                    'work_type': work_type_entry,
+                    'subtitle': work_subtitle}
+                )
+
+            if work_created:
+                logger.debug("Created work '{}' subtitled '{}'.".format(
+                    work_title,
+                    work_subtitle))
+
+            # get work alternative titles or create them
+            if 'alternative_titles' in dict_work:
+                for alt_title in dict_work['alternative_titles']:
+                    self.create_alternative_title(
+                            work_entry,
+                            alt_title)
+
+            # save work in the database
+            work_entry.save()
+
+    def create_alternative_title(self, work_entry, alt_title):
+        """Create work alternative title in the database if necessary."""
+        _, work_alt_title_created = WorkAlternativeTitle.objects.get_or_create(
+                title__iexact=alt_title,
+                work__title__iexact=work_entry.title,
+                work__subtitle__iexact=work_entry.subtitle,
+                defaults={
+                    'title': alt_title,
+                    'work': work_entry}
+                )
+
+        if work_alt_title_created:
+            logger.debug("Created alternative titles '{}' for '{}'.".format(
+                alt_title, work_entry.title))
+
+    def createworks(self):
+        """Create or update works provided."""
+        # parse the work file to get the data structure
+        try:
+            works = self.parser.parse_work(self.work_file)
+        except BaseException as exc:
+            raise CommandError("{}".format(str(exc))) from exc
+
+        work_success = True
+        # get works or create it
+        for worktype_query_name, dict_work_type in works.items():
+
+            logger.debug("Get work type '{}'".format(worktype_query_name))
+            # get work type
+            try:
+                work_type_entry = WorkType.objects.get(
+                        query_name__iexact=worktype_query_name
+                        )
+
+                # get work titles and their attributes
+                for work_title, dict_work in dict_work_type.items():
+
+                    if dict_work is None:
+                        # create an empty dictionnary in the case only
+                        # the title has been provided
+                        dict_work = {}
+                    # check that the work data is well structured
+                    elif not self.check_parser_result(
```

Comment:
I think you should drop the `elif` statement in favor of play `if`. If you provide a default structure, it should pass the structure check. The code would be easier to read.